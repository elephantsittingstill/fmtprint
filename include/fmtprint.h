#ifndef FMTPRINT_H
#define FMTPRINT_H

#include <stdarg.h>

void fmtprint( const char *fstring, ... );

#endif
