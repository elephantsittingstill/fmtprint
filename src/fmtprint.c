#include "internal.h"

void fmtprint( const char *fstring, ... ) {

	va_list args;
	va_start( args, fstring );

#if USE_COMPUTED_GOTO
	parse_format_switch( &fstring, &args );
#else
	parse_format_goto( &fstring, &args );
#endif

	return;
}
