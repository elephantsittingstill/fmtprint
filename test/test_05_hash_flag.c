#include "../include/fmtprint.h"

int main() {
	fmtprint( "Hex with no hash flag: %x\n"
			  "Hex with hash flag: %#x\n",
			  42, 42 );
}
