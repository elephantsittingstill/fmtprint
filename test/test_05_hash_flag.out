Hex with no hash flag: 2a
Hex with hash flag: 0x2a

